scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -846,13 +846,13 @@ def _initialize(self):
         self.__target_pids = []
 
         # convert target pid into a list. Target pids can be a single pid or a CSV of pids
-        if __target_pids:
+        if __target_pids == "$$":
+            # do not resolve $$ to current process id in case the current process forks and changes pid
+            self.__target_pids = "$$"
```

Comment:
It's a little confusing that `self.__target_pids` can be a string in some cases (`$$`) and a list of numbers.

Reply:
makes sense. i'll keep it consistent